diff --git a/abstract/edwards.js b/abstract/edwards.js
index ca506901d86a1d3662227d5ee4ab9ada885c2f45..38f2412bac8c7bf6076921fe32c24b1d585c8eea 100644
--- a/abstract/edwards.js
+++ b/abstract/edwards.js
@@ -4,6 +4,10 @@ exports.PrimeEdwardsPoint = void 0;
 exports.edwards = edwards;
 exports.eddsa = eddsa;
 exports.twistedEdwards = twistedEdwards;
+
+const NativeAvalabsCrypto = require("react-native-nitro-avalabs-crypto");
+const getExtendedPublicKeyFromNativeLib = NativeAvalabsCrypto.getExtendedPublicKey;
+
 /**
  * Twisted Edwards curve. The formula is: ax² + y² = 1 + dx²y².
  * For design rationale of types / exports, see weierstrass module documentation.
@@ -460,6 +464,15 @@ function eddsa(Point, cHash, eddsaOpts = {}) {
     }
     /** Calculates EdDSA pub key. RFC8032 5.1.5. */
     function getPublicKey(secretKey) {
+        const con = console;
+        try {
+            const nativeRes = getExtendedPublicKeyFromNativeLib(secretKey);
+            return nativeRes;
+        } catch (e) {
+            con.log(`Native lib failed, falling back to JS implementation ${e}`);
+            // fall through to JS implementation below
+        }
+
         return getExtendedPublicKey(secretKey).pointBytes;
     }
     // int('LE', SHA512(dom2(F, C) || msgs)) mod N
