diff --git a/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/62/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java b/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/62/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
index 1609813..9c6fefa 100644
--- a/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/62/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
+++ b/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/62/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
@@ -220,7 +220,7 @@ public class ReanimatedNativeHierarchyManager extends NativeViewHierarchyManager
     }
 
     if (initOk) {
-      setLayoutAnimationEnabled(true);
+      // setLayoutAnimationEnabled(true);
     }
   }
 
diff --git a/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java b/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
index a022151..cdc49d8 100644
--- a/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
+++ b/node_modules/react-native-reanimated/android/src/reactNativeVersionPatch/nativeHierarchyManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java
@@ -247,7 +247,7 @@ public class ReanimatedNativeHierarchyManager extends NativeViewHierarchyManager
     }
 
     if (initOk) {
-      setLayoutAnimationEnabled(true);
+      //setLayoutAnimationEnabled(true);
     }
   }
 
