diff --git a/node_modules/react-native-multithreading/cpp/RNMultithreadingInstaller.cpp b/node_modules/react-native-multithreading/cpp/RNMultithreadingInstaller.cpp
index 7a3e6e5..345dcbb 100644
--- a/node_modules/react-native-multithreading/cpp/RNMultithreadingInstaller.cpp
+++ b/node_modules/react-native-multithreading/cpp/RNMultithreadingInstaller.cpp
@@ -5,8 +5,8 @@
 
 #if __has_include(<RNReanimated/Scheduler.h>)
 #include <RNReanimated/Scheduler.h>
-#include <RNReanimated/ShareableValue.h>
 #include <RNReanimated/RuntimeManager.h>
+#include <RNReanimated/ShareableValue.h>
 #include <RNReanimated/RuntimeDecorator.h>
 #include <RNReanimated/ErrorHandler.h>
 #else
