diff --git a/node_modules/react-native-popable/lib/typescript/Popable.d.ts b/node_modules/react-native-popable/lib/typescript/Popable.d.ts
index 8257eba..a87ded1 100644
--- a/node_modules/react-native-popable/lib/typescript/Popable.d.ts
+++ b/node_modules/react-native-popable/lib/typescript/Popable.d.ts
@@ -18,6 +18,7 @@ export declare type PopableProps = {
     style?: PopoverProps['style'];
     visible?: boolean;
     wrapperStyle?: ViewProps['style'];
+    hitslop?: ViewProps['hitslop'];
 };
 declare const Popable: React.ForwardRefExoticComponent<PopableProps & React.RefAttributes<PopableManager>>;
 export default Popable;
diff --git a/node_modules/react-native-popable/lib/typescript/index.d.ts b/node_modules/react-native-popable/lib/typescript/index.d.ts
index ffe74e2..4cf6f33 100644
--- a/node_modules/react-native-popable/lib/typescript/index.d.ts
+++ b/node_modules/react-native-popable/lib/typescript/index.d.ts
@@ -1,4 +1,4 @@
 export { default as Popover } from './Popover';
-export { default as Popable } from './Popable';
+export { default as Popable, PopableProps } from './Popable';
 export { default as usePopable } from './use-popable';
 export * from './use-popable/types';
diff --git a/node_modules/react-native-popable/src/Popable.tsx b/node_modules/react-native-popable/src/Popable.tsx
index 75f214e..33a60a7 100644
--- a/node_modules/react-native-popable/src/Popable.tsx
+++ b/node_modules/react-native-popable/src/Popable.tsx
@@ -7,6 +7,7 @@ import React, {
   useState,
 } from 'react';
 import {
+  Dimensions,
   Platform,
   // @ts-ignore
   Pressable,
@@ -35,6 +36,7 @@ export type PopableProps = {
   style?: PopoverProps['style'];
   visible?: boolean;
   wrapperStyle?: ViewProps['style'];
+  hitslop?: ViewProps['hitSlop'];
 };
 
 const DEFAULT_LAYOUT = {
@@ -44,6 +46,7 @@ const DEFAULT_LAYOUT = {
   y: 0,
 };
 
+const { width: screenWidth } = Dimensions.get('window')
 const Popable = forwardRef<PopableManager, PopableProps>(function Popable(
   {
     action = 'press',
@@ -61,6 +64,7 @@ const Popable = forwardRef<PopableManager, PopableProps>(function Popable(
     style,
     visible,
     wrapperStyle,
+    hitslop
   },
   ref
 ) {
@@ -185,6 +189,12 @@ const Popable = forwardRef<PopableManager, PopableProps>(function Popable(
       case 'top':
       case 'bottom':
         left = (popoverLayout.width - childrenLayout.width) / 2;
+        if (childrenLayout.x < left) {
+          left = 0
+        }
+        if (childrenLayout.x + childrenLayout.width + left > screenWidth) {
+          left = left*2
+        }
         break;
     }
 
@@ -265,6 +275,7 @@ const Popable = forwardRef<PopableManager, PopableProps>(function Popable(
 
       <Pressable
         ref={childrenRef}
+        hitSlop={hitslop}
         onLayout={handleChildrenLayout}
         {...handlers}
       >
diff --git a/node_modules/react-native-popable/src/index.tsx b/node_modules/react-native-popable/src/index.tsx
index ffe74e2..4cf6f33 100644
--- a/node_modules/react-native-popable/src/index.tsx
+++ b/node_modules/react-native-popable/src/index.tsx
@@ -1,4 +1,4 @@
 export { default as Popover } from './Popover';
-export { default as Popable } from './Popable';
+export { default as Popable, PopableProps } from './Popable';
 export { default as usePopable } from './use-popable';
 export * from './use-popable/types';
