diff --git a/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm b/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm
index a00e143..19213c7 100644
--- a/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm
+++ b/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm
@@ -328,7 +328,11 @@ - (void)drawRect:(CGRect)rect
 #endif
   CGContextRef context = UIGraphicsGetCurrentContext();

-  [self drawToContext:context withRect:[self bounds]];
+  @try {
+    [self drawToContext:context withRect:[self bounds]];
+  } @catch (NSException *exception) {
+    RCTLogWarn(@"[RNSVGSvgView drawRect] %@", exception.reason);
+  }
 }

 - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event