diff --git a/node_modules/pino/browser.js b/node_modules/pino/browser.js
index bfdb8ea..52e5269 100644
--- a/node_modules/pino/browser.js
+++ b/node_modules/pino/browser.js
@@ -90,7 +90,7 @@ function pino (opts) {
   }
   if (opts.enabled === false || opts.browser.disabled) opts.level = 'silent'
   const level = opts.level || 'info'
-  const logger = Object.create(proto)
+  const logger = Object.assign({}, proto); // Object.create(proto)
   if (!logger.log) logger.log = noop
 
   setupBaseLogFunctions(logger, levels, proto)