diff --git a/node_modules/react-native-head-tab-view/GestureContainer.tsx b/node_modules/react-native-head-tab-view/GestureContainer.tsx
index ee5affe..ed31ff2 100644
--- a/node_modules/react-native-head-tab-view/GestureContainer.tsx
+++ b/node_modules/react-native-head-tab-view/GestureContainer.tsx
@@ -599,7 +599,7 @@ export default React.forwardRef(GestureContainer)
 const styles = StyleSheet.create({
     container: {
         flex: 1,
-        overflow: 'hidden'
+        overflow: 'visible'
     },
     tabbarStyle: {
         position: 'absolute',
diff --git a/node_modules/react-native-head-tab-view/createCollapsibleScrollView.tsx b/node_modules/react-native-head-tab-view/createCollapsibleScrollView.tsx
index 9ceedc6..dc55c7d 100644
--- a/node_modules/react-native-head-tab-view/createCollapsibleScrollView.tsx
+++ b/node_modules/react-native-head-tab-view/createCollapsibleScrollView.tsx
@@ -48,7 +48,7 @@ const SceneComponent: React.FC<NormalSceneProps & HPageViewProps> = (
     }
 ) => {
     if (onScroll !== undefined) {
-        console.warn("Please do not assign onScroll")
+        // console.warn("Please do not assign onScroll")
     }
     const {
         shareAnimatedValue = useSharedValue(0),
@@ -157,6 +157,7 @@ const SceneComponent: React.FC<NormalSceneProps & HPageViewProps> = (
     const onScrollAnimateEvent = useAnimatedScrollHandler(
         {
             onScroll: (event, ctx) => {
+                onScroll && runOnJS(onScroll)(event)
                 realY.value = event.contentOffset.y
                 let moveY = Math.max(event.contentOffset.y, 0)
 
