diff --git a/node_modules/react-native-webview-crypto/index.js b/node_modules/react-native-webview-crypto/index.js
index ec96971..b52ea51 100644
--- a/node_modules/react-native-webview-crypto/index.js
+++ b/node_modules/react-native-webview-crypto/index.js
@@ -5,14 +5,12 @@ const { MainWorker, webViewWorkerString } = require("webview-crypto");
 
 const styles = StyleSheet.create({
   hide: {
-    display: "none",
-    position: "absolute",
-
-    width: 0,
-    height: 0,
-
-    flexGrow: 0,
-    flexShrink: 1
+    position: 'absolute',
+    top: -1000,
+    left: -1000,
+    width: 1,
+    height: 1,
+    opacity: 0
   }
 });
 
