
diff --git a/node_modules/react-native-reanimated/src/hook/useAnimatedSensor.ts b/node_modules/react-native-reanimated/src/hook/useAnimatedSensor.ts
index 6256687..2d6e16e 100644
--- a/node_modules/react-native-reanimated/src/hook/useAnimatedSensor.ts
+++ b/node_modules/react-native-reanimated/src/hook/useAnimatedSensor.ts
@@ -88,14 +88,17 @@ function adjustVectorToInterfaceOrientation(data: Value3D) {
  */
 export function useAnimatedSensor(
   sensorType: SensorType.ROTATION,
+  isActive: boolean,
   userConfig?: Partial<SensorConfig>
 ): AnimatedSensor<ValueRotation>;
 export function useAnimatedSensor(
   sensorType: Exclude<SensorType, SensorType.ROTATION>,
+  isActive: boolean,
   userConfig?: Partial<SensorConfig>
 ): AnimatedSensor<Value3D>;
 export function useAnimatedSensor(
   sensorType: SensorType,
+  isActive: boolean,
   userConfig?: Partial<SensorConfig>
 ): AnimatedSensor<ValueRotation> | AnimatedSensor<Value3D> {
   const userConfigRef = useRef(userConfig);
@@ -144,18 +147,22 @@ export function useAnimatedSensor(
     const adjustToInterfaceOrientation =
       ref.current.config.adjustToInterfaceOrientation;
 
-    const id = registerSensor(sensorType, config, (data) => {
-      'worklet';
-      if (adjustToInterfaceOrientation) {
-        if (sensorType === SensorType.ROTATION) {
-          data = adjustRotationToInterfaceOrientation(data as ValueRotation);
-        } else {
-          data = adjustVectorToInterfaceOrientation(data as Value3D);
+    let id = -1
+
+    if (isActive) {
+      id = registerSensor(sensorType, config, (data) => {
+        'worklet';
+        if (adjustToInterfaceOrientation) {
+          if (sensorType === SensorType.ROTATION) {
+            data = adjustRotationToInterfaceOrientation(data as ValueRotation);
+          } else {
+            data = adjustVectorToInterfaceOrientation(data as Value3D);
+          }
         }
-      }
-      sensorData.value = data;
-      callMicrotasks();
-    });
+        sensorData.value = data;
+        callMicrotasks();
+      });
+    }
 
     if (id !== -1) {
       // if sensor is available
@@ -172,7 +179,13 @@ export function useAnimatedSensor(
     return () => {
       ref.current.unregister();
     };
-  }, [sensorType, config]);
+  }, [sensorType, config, isActive]);
+
+  useEffect(() => {
+    if (!isActive) {
+      ref.current.unregister();
+    }
+  }, [isActive, ref]);
 
   return ref.current as AnimatedSensor<ValueRotation> | AnimatedSensor<Value3D>;
 }
