diff --git a/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm b/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm
index 9306b04..8e6d73b 100644
--- a/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm
+++ b/node_modules/react-native-svg/apple/Elements/RNSVGSvgView.mm
@@ -290,8 +290,12 @@ - (void)drawRect:(CGRect)rect
   }
   _boundingBox = rect;
   CGContextRef context = UIGraphicsGetCurrentContext();
-
-  [self drawToContext:context withRect:rect];
+  @try {
+    [self drawToContext:context withRect:rect];
+  }
+  @catch (NSException *exception) {
+    RCTLogWarn(@"[RNSVGSvgView drawRect] %@", exception.reason);
+  }
 }
 
 - (RNSVGPlatformView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
