diff --git a/ReactCommon/jsi/JSCRuntime.cpp b/ReactCommon/jsi/JSCRuntime.cpp
index 5d8487b2cd5b93b5238df0fb547e44c62fefc0b4..f960c6fc26d602bf3a83d4e4305a4efd64e0381a 100644
--- a/ReactCommon/jsi/JSCRuntime.cpp
+++ b/ReactCommon/jsi/JSCRuntime.cpp
@@ -392,12 +392,12 @@ JSCRuntime::~JSCRuntime() {
   // has started.
   ctxInvalid_ = true;
   JSGlobalContextRelease(ctx_);
-#ifndef NDEBUG
-  assert(
-      objectCounter_ == 0 && "JSCRuntime destroyed with a dangling API object");
-  assert(
-      stringCounter_ == 0 && "JSCRuntime destroyed with a dangling API string");
-#endif
+// #ifndef NDEBUG
+//   assert(
+//       objectCounter_ == 0 && "JSCRuntime destroyed with a dangling API object");
+//   assert(
+//       stringCounter_ == 0 && "JSCRuntime destroyed with a dangling API string");
+// #endif
 }
 
 std::shared_ptr<const jsi::PreparedJavaScript> JSCRuntime::prepareJavaScript(
diff --git a/index.js b/index.js
index d59ba3454e9cabed1231554418b5b288c2461a0b..3f51111c619952c12178445bc3fb89cca8eaa4e6 100644
--- a/index.js
+++ b/index.js
@@ -456,11 +456,12 @@ module.exports = {
     );
   },
   get ViewPropTypes(): $FlowFixMe {
-    invariant(
-      false,
-      'ViewPropTypes has been removed from React Native. Migrate to ' +
-        "ViewPropTypes exported from 'deprecated-react-native-prop-types'.",
-    );
+    return require('deprecated-react-native-prop-types').ViewPropTypes
+    // invariant(
+    //   false,
+    //   'ViewPropTypes has been removed from React Native. Migrate to ' +
+    //     "ViewPropTypes exported from 'deprecated-react-native-prop-types'.",
+    // );
   },
 };
 
diff --git a/scripts/react-native-xcode.sh b/scripts/react-native-xcode.sh
index 927ec76c3adc1fad91acfd97c8653e69493b6b0a..6e4043a5956c5d652a668747eb4d5d5a8f4f9904 100755
--- a/scripts/react-native-xcode.sh
+++ b/scripts/react-native-xcode.sh
@@ -129,7 +129,7 @@ fi
 
 PACKAGER_SOURCEMAP_FILE=
 if [[ $EMIT_SOURCEMAP == true ]]; then
-  if [[ $USE_HERMES != false ]]; then
+  if [[ $USE_HERMES == true ]]; then
     PACKAGER_SOURCEMAP_FILE="$CONFIGURATION_BUILD_DIR/$(basename $SOURCEMAP_FILE)"
   else
     PACKAGER_SOURCEMAP_FILE="$SOURCEMAP_FILE"
@@ -138,7 +138,7 @@ if [[ $EMIT_SOURCEMAP == true ]]; then
 fi
 
 # Hermes doesn't require JS minification.
-if [[ $USE_HERMES != false && $DEV == false ]]; then
+if [[ $USE_HERMES == true && $DEV == false ]]; then
   EXTRA_ARGS="$EXTRA_ARGS --minify false"
 fi
 
@@ -153,7 +153,7 @@ fi
   $EXTRA_ARGS \
   $EXTRA_PACKAGER_ARGS
 
-if [[ $USE_HERMES == false ]]; then
+if [[ $USE_HERMES != true ]]; then
   cp "$BUNDLE_FILE" "$DEST/"
   BUNDLE_FILE="$DEST/main.jsbundle"
 else
