diff --git a/node_modules/web3-providers-http/lib/index.js b/node_modules/web3-providers-http/lib/index.js
index ba5c9ae..be031b5 100644
--- a/node_modules/web3-providers-http/lib/index.js
+++ b/node_modules/web3-providers-http/lib/index.js
@@ -27,7 +27,7 @@ var http = require('http');
 var https = require('https');
 // Apply missing polyfill for IE
 require('cross-fetch/polyfill');
-require('es6-promise').polyfill();
+// require('es6-promise').polyfill();
 require('abortcontroller-polyfill/dist/polyfill-patch-fetch');
 /**
  * HttpProvider should be used to send rpc calls over http
